pkg/filetype/file_type.go

```diff
@@ -12,111 +12,139 @@ import (
 type FileType struct {
 	Name       string
 	Extensions map[string]struct{}
+	KnownFiles map[string]struct{}
 	Validator  validator.Validator
 }
 
 // Instance of the FileType object to
 // represent a JSON file
 var JSONFileType = FileType{
-	"json",
-	tools.ArrToMap("json"),
-	validator.JSONValidator{},
+	Name:       "json",
+	Extensions: tools.ArrToMap("json"),
+	Validator:  validator.JSONValidator{},
 }
 
 // Instance of the FileType object to
 // represent a YAML file
 var YAMLFileType = FileType{
-	"yaml",
-	tools.ArrToMap("yml", "yaml"),
-	validator.YAMLValidator{},
+	Name:       "yaml",
+	Extensions: tools.ArrToMap("yml", "yaml"),
+	KnownFiles: tools.ArrToMap(
+		".clang-format",
+		".clang-tidy",
+		".clangd",
+		".gemrc",
+		"glide.lock",
```

Comment:
What would be your call here @kehoecj about the lock files.

I added them as they are known files, but is that the tool responsibility to validate a lock file that is often automatically generated by the tool that uses it.

Reply:
>but is that the tool responsibility to validate a lock file that is often automatically generated by the tool that uses it.

Agree - not sure there is a lot of benefit to adding lock files